pandas/_libs/tslibs/offsets.pyx

```diff
@@ -4847,53 +4847,36 @@ cpdef to_offset(freq, bint is_period=False):
-                    name = c_OFFSET_DEPR_FREQSTR[name.upper()]
                 if (not is_period and
                         name != name.upper() and
                         name.lower() not in {"s", "ms", "us", "ns"} and
                         name.upper().split("-")[0].endswith(("S", "E"))):
-                    warnings.warn(
-                        f"\'{name}\' is deprecated and will be removed "
-                        f"in a future version, please use "
-                        f"\'{name.upper()}\' instead.",
-                        FutureWarning,
-                        stacklevel=find_stack_level(),
-                    )
-                    name = name.upper()
-                if is_period and name.upper() in c_REVERSE_OFFSET_DEPR_FREQSTR:
-                    if name.upper().startswith("Y"):
+                    if name.upper() in c_OFFSET_TO_PERIOD_FREQSTR:
                         raise ValueError(
-                            f"for Period, please use \'Y{name.upper()[2:]}\' "
-                            f"instead of \'{name}\'"
+                            f"\'{name}\' is no longer supported for offsets. Please "
+                            f"use \'{name.upper()}\' instead."
                         )
-                    if (name.upper().startswith("B") or
-                            name.upper().startswith("S") or
-                            name.upper().startswith("C")):
+                    else:
+                        raise ValueError(INVALID_FREQ_ERR_MSG.format(name))
+                if (is_period and
+                        name.upper().split("-")[0].endswith(("E")) and
+                        name.upper() in c_OFFSET_TO_PERIOD_FREQSTR):
+                    if name.upper().startswith("B"):
                         raise ValueError(INVALID_FREQ_ERR_MSG.format(name))
                     else:
                         raise ValueError(
-                            f"for Period, please use "
-                            f"\'{c_REVERSE_OFFSET_DEPR_FREQSTR.get(name.upper())}\' "
-                            f"instead of \'{name}\'"
+                            f"{name} is not supported as period frequency"
                         )
-                elif is_period and name.upper() in c_OFFSET_DEPR_FREQSTR:
+
+                elif is_period and name.upper() in c_OFFSET_REMOVED_FREQSTR:
                     if name.upper() != name:
-                        warnings.warn(
-                            f"\'{name}\' is deprecated and will be removed in "
-                            f"a future version, please use \'{name.upper()}\' "
-                            f"instead.",
-                            FutureWarning,
-                            stacklevel=find_stack_level(),
-                        )
-                    name = c_OFFSET_DEPR_FREQSTR.get(name.upper())
+                        raise ValueError(INVALID_FREQ_ERR_MSG.format(name))
+                    name = c_OFFSET_REMOVED_FREQSTR.get(name.upper())
```

Comment:
Are you sure about this line? It looks very odd to me to do

```
                if is_period and name.upper() in c_OFFSET_REMOVED_FREQSTR:
                    name = c_OFFSET_REMOVED_FREQSTR.get(name.upper())
```

If you need to map `'M'` (period) to its associated offset alias ('ME'), then it's there another dictionary, or another way, to do that? `c_OFFSET_REMOVED_FREQSTR` will presumably be removed at some point

Reply:
It looks to me that `c_OFFSET_REMOVED_FREQSTR` is functioning as both:
- aliases which used to be valid for offsets, but have been renamed
- period aliases, to get the corresponding offset aliases

How about creating another dictionary, which maps periods aliases to their corresponding offset aliases? I think this one could be a lot smaller - for example, it wouldn't need `"BY": "BYE",`, right?

Then, `to_offset` could be a lot simpler, something like
```python
if not is_period and name in c_OFFSET_REMOVED_FREQSTR:
    # raise error message
if is_period:
    if name in c_PERIOD_TO_OFFSET_FREQSTR:
        name = c_PERIOD_TO_OFFSET_FREQSTR[name]
    else:
        # raise: `name` not support for period
```
?

Reply:
> Are you sure about this line? It looks very odd to me to do
> ```
>                 if is_period and name.upper() in c_OFFSET_REMOVED_FREQSTR:
>                     name = c_OFFSET_REMOVED_FREQSTR.get(name.upper())
> ```
> If you need to map `'M'` (period) to its associated offset alias ('ME'), then it's there another dictionary, or another way, to do that? `c_OFFSET_REMOVED_FREQSTR` will presumably be removed at some point

thanks for the comment. It seems odd, I agree. I removed this check.